### PR TITLE
Use Google Tag Manager for analytics

### PR DESCRIPTION
**Pull request by jonasfj**
We probably still need to migrate the generated dartdoc, but
I don't think we can make that migration experimental.

### CODE REVIEW

File: app/lib/frontend/templates/views/shared/layout.mustache [hunk 1]
```diff
@@ -4,8 +4,15 @@
 <!DOCTYPE html>
 <html lang="en-us">
 <head>
+  {{#is_experimental}}
+  <script async="async" src="https://www.googletagmanager.com/gtm.js?id=GTM-MX6DBN9"></script>
```
Comment (isoos) [inline app/lib/frontend/templates/views/shared/layout.mustache:8]: Nit: maybe put only the id behind the {{#is_experimental}} condition?

File: static/js/gtm.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+window.dataLayer = window.dataLayer || [];
```
Comment (isoos) [inline static/js/gtm.js:1]: We should probably put a license header here (+ on gtag.js)

File: static/js/gtm.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+window.dataLayer.push({
```
Comment (isoos) [inline static/js/gtm.js:2]: switch the order?

```
function gtag() { dataLayer.push(arguments); }
gtag({
  'gtm.start': new Date().getTime(),
  event: 'gtm.js'
});
```

**Review by isoos — APPROVED** [review]
lgtm, only a few nits.

File: static/js/gtm.js (hunk 3 above)
Comment (jonasfj) [reply static/js/gtm.js:2]: I don't think that is the same.. `arguments` behaves as an array..

Comment (isoos) [reply static/js/gtm.js:2]: /facepalm

File: app/lib/frontend/templates/views/shared/layout.mustache (hunk 1 above)
Comment (jonasfj) [reply app/lib/frontend/templates/views/shared/layout.mustache:8]: It's more than the ID, it's a different script... I don't know if the contents is different, but it is a different script :D

File: static/js/gtm.js (hunk 2 above)
Comment (jonasfj) [reply static/js/gtm.js:5]: Good idea :D

**Pull request merged**